cmd/kubeadm/app/cmd/init.go

```diff
@@ -151,6 +158,18 @@ func RunInit(out io.Writer, cmd *cobra.Command, args []string, s *kubeadmapi.Kub
 		return err
 	}
 
+	// TODO: Shut off glog info level here:
+	// This call doesn't actually die if it cannot connect... check the version
+	// and see if we get an error:
+	dockerClient := dockertools.ConnectToDockerOrDie(dockerEndpoint, dockerClientTimeout.Duration)
+	if _, err := dockerClient.Version(); err != nil {
```

Comment:
Here would be a good place to check the docker version and output it along with a comment.
If docker == 1.11.2, say "good! you have the right docker version"
if 1.10 or 1.12 say that it isn't the recommended/fully validated version, but it should work fine
if 1.9, fail as k8s doesn't support it anymore (I think)